### DISCUSSION TITLE
Upgrade 12.5.8 - Warning: treating it as an external dependency

### DISCUSSION THREAD

**Discussion by Fasani** [nrwl/nx#6376] [General]
When I upgraded this morning a new property was added to my `nx.json`:

```
"targetDependencies": {
    "build": [
        {
            "target": "build",
            "projects": "dependencies"
        }
    ]
}
```

Now when building my main apps, NX seems to build some of my libraries also, which I would imagine is connected to the above. Now I see multiple messages like this during the build.

`@company-name/shared' is imported by libs/components/src/lib/customer-order-summary/customer-order-summary.component.tsx, but could not be resolved – treating it as an external dependency`

Is this to be expected and should I prevent these warnings somehow? I do not publish my libs externally I just use them for code sharing within my multiple apps, however, I do quite like the building step as it help me spot some unseen issues.

Thank you

**Comment by devzom** [comment]
Having same issue and don't know what to handle it. Any suggestions?
@Fasani you've found smth?
